core/lib/Thelia/Action/Export.php

```diff
@@ -58,6 +61,77 @@ protected function cacheClear(EventDispatcherInterface $dispatcher)
     }
 
     /**
+     * Before a CSV Export
+     *
+     * Replace labels into heading row
+     * Add a heading row with translated column names manually
+     *
+     * Of course the export has to use a CSVFormatter.
+     *
+     * @param ImportExport $event
+     */
+    public function addHeadingRow(ImportExport $event)
+    {
+        $handler = $event->getHandler();
+        $formatter = $event->getFormatter();
+
+        if ($formatter instanceof CSVFormatter) {
+            // Get existing data
+            $formatterData = $event->getData();
+            $data = $formatterData->getData();
+
+            // Get heading labels
+            $heading = $handler->getTranslatedHeading();
+
+            // Complete heading row (with all keys)
+            // - Use label if possible
+            // - Use alias else
+            $orderCols = array_keys($formatterData->getRow(0));
+            $orderCols = array_combine($orderCols, $orderCols);
```

Comment:
```FormatterData::getRow``` may return false if the array is empty.
I'm not sure that ```array_combine``` really likes booleans :)